deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -1863,7 +1863,7 @@ def step(self, closure=None):
                 #    bit16_partitions[partition_id].data.copy_(fp32_partition.data)
                 bit16_partitions = self.parallel_partitioned_bit16_groups[i]
                 fp32_partition = self.single_partition_of_fp32_groups[i]
-                bit16_partitions[partition_id].data.copy_(fp32_partition.data)
+                bit16_partitions[partition_id].data.copy_(fp32_partition.to('cuda').data)
```

Comment:
We should use `get_accelerator()` instead of `cuda` references. 
```suggestion
                bit16_partitions[partition_id].data.copy_(fp32_partition.to(get_accelerator().current_device_name()).data)
```